sdccc/src/test/java/com/draeger/medical/sdccc/manipulation/GRpcManipulationsTest.java

```diff
@@ -88,15 +94,154 @@ void setUp() throws IOException {
+     */
+    @Test
+    @TestDescription("Verifies that a ManipulationResponse with an Optional is deserialized correctly")
+    public void testCreateContextStateWithAssociationDeserialization() {
+
+        final ManipulationResponse<Optional<String>> originalResponse =
+                ManipulationResponse.from(ResponseTypes.Result.RESULT_SUCCESS, Optional.of("testContextStateHandle"));
+
+        final Gson gson = new GsonBuilder()
+                .registerTypeAdapterFactory(new OptionalTypeAdapterFactory())
+                .create();
+        final GsonManipulationSerializer serializer = new GsonManipulationSerializer(gson);
+
+        final Type type = new TypeToken<ManipulationResponse<Optional<String>>>() {}.getType();
+
+        final String json = serializer.serialize(originalResponse);
+        final ManipulationResponse<Optional<String>> deserializedResponse = serializer.deserialize(json, type);
+
+        assertEquals(ResponseTypes.Result.RESULT_SUCCESS, deserializedResponse.getResult());
+        assertTrue(deserializedResponse.getResponse().isPresent());
+        assertEquals(
+                "testContextStateHandle", deserializedResponse.getResponse().get());
+    }
+
+    /**
+     * Verifies that a ManipulationResponse with an empty Optional is serialized correctly.
+     */
+    @Test
+    @TestDescription("Verifies that a ManipulationResponse with an empty Optional is serialized correctly")
+    public void testCreateContextStateWithEmptyOptionalSerialization() {
+
+        final ManipulationResponse<Optional<String>> fallbackResponse =
+                ManipulationResponse.from(ResponseTypes.Result.RESULT_SUCCESS, Optional.empty());
+        when(fallback.createContextStateWithAssociation(anyString(), any())).thenReturn(fallbackResponse);
+
+        final ManipulationResponse<Optional<String>> response =
+                manipulations.createContextStateWithAssociation("dummyHandle", ContextAssociation.ASSOC);
+
+        assertEquals(ResponseTypes.Result.RESULT_SUCCESS, response.getResult());
+        assertTrue(response.getResponse().isEmpty());
+    }
+
+    /**
+     * Verifies that a ManipulationResponse with an empty Optional is deserialized correctly.
+     */
+    @Test
+    @TestDescription("Verifies that a ManipulationResponse with an empty Optional is deserialized correctly")
+    public void testCreateContextStateWithEmptyOptionalDeserialization() {
+
+        final ManipulationResponse<Optional<String>> originalResponse =
+                ManipulationResponse.from(ResponseTypes.Result.RESULT_SUCCESS, Optional.empty());
+
+        final Gson gson = new GsonBuilder()
+                .serializeNulls()
+                .registerTypeAdapterFactory(new OptionalTypeAdapterFactory())
+                .create();
+        final GsonManipulationSerializer serializer = new GsonManipulationSerializer(gson);
+
+        final Type type = new TypeToken<ManipulationResponse<Optional<String>>>() {}.getType();
```

Comment:
I think this would have to be made useable with and used for the inline deserialize function in the companion object of the ManipulationResponse data class in com.draeger.medical.sdccc.manipulation.ManipulationResponse .